src/os_unix.c

```diff
@@ -913,6 +913,22 @@ catch_sigint SIGDEFARG(sigarg)
 }
 #endif
 
+#if defined(SIGUSR1)
+char_u *SIGUSR1_HANDLER_NAME = (char_u *)"SIGUSR1_handler";
+
+    static RETSIGTYPE
+call_sigusr1_hander SIGDEFARG(sigarg)
+{
+
+# if defined(FEAT_USR_CMDS)
+    typval_T rettv;
+    call_vim_function(SIGUSR1_HANDLER_NAME, 0, NULL, FALSE, TRUE, &rettv);
+    clear_tv(&rettv);
```

Comment:
This can't work as-is.  You're not allowed to use dynamically allocate/free memory in a signal handle among plenty of other things. Signals are error prone.  A signal handler should be re-entrant and merely set a volatile a variable and not much more.   That volatile variable can then be checked later when dispatching events outside of the signal handler.

See: https://www.gnu.org/software/libc/manual/html_node/Nonreentrancy.html